### PR TITLE
Fixed short circuit condition to correctly detect @RequestMapping shortcuts #1201

### PR DESCRIPTION
**Pull request by Infrag**
@GetMapping, @PostMaping etc.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@Infrag Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-netflix&pullRequestId=1586)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-netflix&pullRequestId=1586) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/faq) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@Infrag Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-netflix&pullRequestId=1586)!

**Comment by Infrag** [comment]
@spencergibb

File: spring-cloud-netflix-core/src/test/java/org/springframework/cloud/netflix/feign/support/SpringMvcContractTests.java [hunk 1]
```diff
@@ -388,7 +425,7 @@ TestObject getSpecificTest(@PathVariable("classId") String classId,
 		@RequestMapping(path = "/test/{id}", method = RequestMethod.PUT, produces = MediaType.APPLICATION_JSON_VALUE)
 		ResponseEntity<TestObject> getTest(@RequestHeader("Authorization") String auth,
 				@PathVariable("id") String id, @RequestParam("amount") Integer amount);
-
+
```
Comment (ryanjbaxter) [inline spring-cloud-netflix-core/src/test/java/org/springframework/cloud/netflix/feign/support/SpringMvcContractTests.java]: Fix spacing

**Comment by spencergibb** [comment]
Closed with some polish via 66da2d8f48fed92fbff5efadd238b12167deabaa

**Pull request closed without merging**